core-libs/setup/ssr/engine/cx-express-engine.ts

```diff
@@ -0,0 +1,91 @@
+import { StaticProvider } from '@angular/core';
+import {
+  CommonEngine,
+  CommonEngineOptions,
+  CommonEngineRenderOptions,
+} from '@angular/ssr';
+import { Request, Response } from 'express';
+import { REQUEST, RESPONSE } from '../tokens/express.tokens';
+
+export type CxSetupOptions = Readonly<
+  CommonEngineRenderOptions & CommonEngineOptions
+>;
+
+/*
+ * Copyright (c) 2010-2023 Google LLC. http://angular.io/license
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+function getReqResProviders(req: Request, res?: Response): StaticProvider[] {
+  const providers: StaticProvider[] = [
+    {
+      provide: REQUEST,
+      useValue: req,
+    },
+  ];
+  if (res) {
+    providers.push({
+      provide: RESPONSE,
+      useValue: res,
+    });
+  }
+
+  return providers;
+}
+
+export interface CxRenderOptions extends CommonEngineRenderOptions {
+  req: Request;
+  res?: Response;
+}
+
+export function cxExpressEngine(
+  setupOptions: Readonly<CommonEngineRenderOptions & CommonEngineOptions>
+) {
+  const engine = new CommonEngine({
+    bootstrap: setupOptions.bootstrap,
+    providers: setupOptions.providers,
+    enablePerformanceProfiler: setupOptions.enablePerformanceProfiler,
+  });
```

Comment:
[MINOR]
Do you think it would it be more beneficial to use a spread operator to pass all props? I believe it would be more future-proof in case Angular adds or removes something?

Reply:
I'm not sure whether we should do it. `setupOptions` contains parameters that are not used for `CommonEngine` instantiation.